pkg/agent/core/metrics.go

```diff
@@ -0,0 +1,182 @@
+package core
+
+// Abstraction over the []metrics -> CU "algorithm"
+//
+// In general, there's only *really* one ScalingAlgorithm implementation, but we keep it generic so
+// that tests can dependency-inject an alternate implementation that's more
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/neondatabase/autoscaling/pkg/api"
+	"github.com/neondatabase/autoscaling/pkg/util"
+)
+
+type Metrics struct {
+	LoadAverage1Min  float32
+	LoadAverage5Min  float32 // unused. To be removed when api.Metrics.LoadAverage5Min is removed.
+	MemoryUsageBytes float32
+}
+
+// ToAPI converts the Metrics to the subset used for communications with the scheduler plugin and
```

Comment:
Why can't we have only `api.Metrics`? Feels like introducing separate structures will introduce confusion. Is it because multiple components depend on API so it is hard to update?

Reply:
In the future, we will probably have only 1 or 2 fields in `api.Metrics`, but many fields in `core.Metrics`. And yeah, it's annoying to have something that _should_ be an implementation detail of the autoscaler-agent (`core.Metrics`) also affect backwards-compatibility of communications between components.